backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -744,9 +884,7 @@ public FutureExperimentRunDAO(
                 executor);
 
     final var futureCount =
-        InternalFuture.sequence(
-                Arrays.asList(futureLocalContext, futurePredicatesContext, futureProjectIds),
-                executor)
+        InternalFuture.sequence(queryFilterContextList, executor)
```

Comment:
This doesn't require any sorting context.

Reply:
it is not for sorting, i have some condition based InternalFuture so i have added list to keep all future and then added list in `InternalFuture.sequence`

Reply:
You are adding the sorting context. You don't need it.

https://github.com/VertaAI/docs/blob/main/engineering-practices/api-implementation/performance.md#avoid-layers-of-abstraction

Reply:
Yes i realize that and removed it now